lib/srv/ctx.go

```diff
@@ -725,6 +725,15 @@ func getPAMConfig(c *ServerContext) (*PAMConfig, error) {
 
 			result, err := expr.Interpolate(traits)
 			if err != nil {
+				// If the trait isn't passed by the IdP due to misconfiguration
+				// we fallback to setting a value which will indicate this.
+				if trace.IsNotFound(err) {
+					trait := expr.Name()
+					log.Warnf("Attempted to interpolate custom PAM environment with external trait %q but received SAML response does not contain claim %q.", trait, trait)
+					environment[key] = "TELEPORT_DID_NOT_FIND_TRAIT"
```

Comment:
@xacrimon @russjones Why not just skip over non-existent claim, just log it and not insert a placeholder? This is what we do in other places when interpolating role templates IIRC e.g. see https://github.com/gravitational/teleport/blob/v7.0.0-dev.9/lib/services/role.go#L404.

Reply:
Seems better to be consistent then. Adjusted